src/compiler/symbolWalker.ts

```diff
@@ -171,8 +171,8 @@ namespace ts {
                 }
                 const t = getTypeOfSymbol(symbol);
                 visitType(t); // Should handle members on classes and such
-                if (symbol.flags & SymbolFlags.HasExports) {
-                    symbol.exports!.forEach(visitSymbol);
+                if (symbol.exports) {

```

Comment:
so when does it have the flag but no exports? this is more concerning than the error really.

Reply:
We don't have a repro for this unfortunately.

Reply:
we only use `HasExports` in two places other than this one.. once in the binder and once
in the checker..  I wounder if we can just remove it all together, and check `exports` instead.

Reply:
@weswigham and @sandersn thoughts?

Reply:
`HasExports` afaik is set on container symbols which _can_ have exports, while the member is present if there actually are any. I imagine the usage in the checker is likely incorrect - AFAIK it should only be used in the binder to determine how to initialize a symbol.

Reply:
It's only an alias, so even in the binder, the code might actually be clearer without it. (or with a comment instead)